src/controller/CHIPDeviceController.cpp

```diff
@@ -905,19 +908,73 @@ void DeviceCommissioner::OnDiscoveredDeviceOverBleError(void * appState, CHIP_ER
 #endif // CONFIG_NETWORK_LAYER_BLE
 
 #if CHIP_DEVICE_CONFIG_ENABLE_WIFIPAF
+void DeviceCommissioner::SetChkObjValid(void * appObj, ObjChkAction action, bool * pIsObjValid)
+{
+    static std::vector<void *> ObjVector;
+    bool IsObjValid = false;
+
+    switch (action)
+    {
+    case ObjChkAction::Set: {
+        for (auto lt = ObjVector.begin(); lt != ObjVector.end(); lt++)
+        {
+            if (*lt == appObj)
+            {
+                IsObjValid = true;
+                break;
+            }
+        }
+        if (IsObjValid == false)
+        {
+            ObjVector.push_back(appObj);
+            IsObjValid = true;
+        }
+    }
+    break;
+    case ObjChkAction::Check: {
+        for (auto lt = ObjVector.begin(); lt != ObjVector.end(); lt++)
+        {
+            if (*lt == appObj)
```

Comment:
OK, so if an object got deallocated and another one got allocated at the same memory address, this check will pass, yes?  So we could be getting a callback for totally the wrong thing (or on totally the wrong object, depending on how you want to think about it).

Reply:
Yes, I will change it